### PR TITLE
Reduce memory consumption in batched_forward_pass

### PR DESCRIPTION
**Pull request by ohashi56225**
This PR reduces memory consumption in `batched_forward_pass` of PPOTrainer, by avoiding the storage of logits when they are not necessary.

Before this PR, `batched_forward_pass` stored all of the model's `logits` all the time like other tensors such as `values` and `logprobs`. Here, `logits` tensors have a much larger size (batch_size * tokens * **vocabulary_size**) compared to `logprobs` and `values` tensors (batch_size × tokens), consuming a significant amount of cuda memory.

I have modified `batched_forward_pass` to avoid unnecessary storage of `logits`, which is only required when calculating entropy in the `loss` method.

### CODE REVIEW

File: trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -644,6 +644,7 @@ def batched_forward_pass(
         queries: torch.Tensor,
         responses: torch.Tensor,
         model_inputs: dict,
+        return_logits: bool = False,
```
Comment (younesbelkada) [inline trl/trainer/ppo_trainer.py:647]: Could you add a sentence stating what this arg does below on the docstring

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for fixing and for taking care of the memory consumption
This looks very good to me!
Would love to hear @lvwerra 's thoughts here

File: trl/trainer/ppo_trainer.py (hunk 1 above)
Comment (ohashi56225) [reply trl/trainer/ppo_trainer.py:647]: Sure!
How about this sentence as the docstring text?

```
return_logits (`bool`, *optional*, defaults to `False`):
    Whether to return all_logits. Set to `False` if logits are not needed to reduce memory consumption.
```

Comment (younesbelkada) [reply trl/trainer/ppo_trainer.py:647]: Awesome, this looks great!

Comment (ohashi56225) [reply trl/trainer/ppo_trainer.py:647]: Thanks, I added the docstring!

Comment (younesbelkada) [reply trl/trainer/ppo_trainer.py:647]: Great thank you!

**Review by lvwerra — APPROVED** [review]
Looks great, thanks!
